pages/maker/apps/index.js

```diff
@@ -11,131 +11,115 @@ import { MakerContainer, MakerContentBox, MakerStickyStatusBox } from '@componen
-    return id
-  }
+    const updateMakerRoute = (id) =>
+      router.push('/maker/apps', `/maker/apps/${id}`, {
+        shallow: true
+      })
 
-  useEffect(() => {
-    async function fetchAppsOnInit () {
-      await fetchApps({ apiServer, user })(dispatch)
-      if (hasAppId) {
-        dispatch(selectAppAction(getAppId()))
-      }
-      if (!hasAppId && process.browser) {
-        setTimeout(() => {
-          console.log(appList)
-          const { id } = appList[0]
-          dispatch(selectAppAction(id))
-        })
-      }
-    }
-    fetchAppsOnInit()
-  }, [])
+    if (maker.loading || !selectedApp) return <LoadingPage />
 
-  if (maker.loading || !selectedApp) {
     return (
       <Page innerPadding={0} wrap>
-        <Flex>
-          <Box width={1}>
-            <Type fontSize={5} my={7} textAlign="center">{maker.loading ? 'Loading...' : errorMessage}</Type>
+        <Head title={selectedApp.name} />
+        <MakerContainer>
+          <Box>
+            {appList.length && (
+              <select onChange={(e) => handleChangingApp(e, updateMakerRoute)(dispatch)} value={appId}>
+                {appList.map(({ name, id }) => (
+                  <option key={id} value={id}>
+                    {name}
+                  </option>
+                ))}
+              </select>
+            )}
           </Box>
-        </Flex>
+          <Type fontSize={3} fontWeight={500} mx={[4, 6]} py={6} px={[20, 0]}>
+            {selectedApp.name}
+          </Type>
+          <Flex
+            flexDirection={['column', 'column', 'row-reverse']}
+            maxWidth={['unset', 'unset', 1140]}
+            alignItems="flex-start"
+          >
+            <MakerStickyStatusBox>
+              <Maker.Status app={selectedApp} apiServer={apiServer} status={completionStatus} />
+            </MakerStickyStatusBox>
+            <Box>
+              <MakerContentBox>
+                <Maker.PaymentDetails app={selectedApp} user={user} apiServer={apiServer} dispatch={dispatch} />
+              </MakerContentBox>
+              <MakerContentBox>
+                <Maker.KYC app={selectedApp} user={user} apiServer={apiServer} />
+              </MakerContentBox>
+              <MakerContentBox>
+                <Maker.ParticipationAgreement app={selectedApp} user={user} apiServer={apiServer} />
+              </MakerContentBox>
+            </Box>
+          </Flex>
+        </MakerContainer>
       </Page>
     )
   }
+)
 
-  function handleChangingApp (event) {
-    event.persist()
-    const id = event.target.value
-    dispatch(selectAppAction(id))
-    updateMakerRoute(id)
+MakerPortal.getInitialProps = async ({ req, reduxStore, ...ctx }) => {
+  const { params, universalCookies } = req
+  const userCookie = universalCookies.cookies.jwt
+  const appId = getAppId(params)
+  const apiServer = selectApiServer(reduxStore.getState())
+  await fetchApps({ apiServer, user: { jwt: userCookie } })(reduxStore.dispatch)
```

Comment:
Now that this action is being dispatched on next's side, I'm never making it past the Loading screen. Is this the issue you were referring to where the initial state is being overwritten by the client?